pip_api/_parse_requirements.py

```diff
@@ -172,6 +176,14 @@ def _url_to_path(url):
     return path
 
 
+class Requirement(requirements.Requirement):
+    def __init__(self, *args, **kwargs):
+        self.hashes = kwargs.pop("hashes", None)
+
+        super().__init__(*args, **kwargs)
```

Comment:
N.B.: `requirements.Requirement` doesn't have any state/API for hashes, so I created a thin child wrapper instead. I figured this would be the least invasive approach, since it doesn't cause any API breakage, but let me know if there's a better/preferred alternative.